gogio/iosbuild.go

```diff
@@ -164,31 +166,53 @@ func exeIOS(tmpDir, target, app string, bi *buildInfo) error {
 		return err
 	}
 	mainm := filepath.Join(tmpDir, "main.m")
-	const mainmSrc = `@import UIKit;
+	mainmSrc := struct {
+		Deeplink []string
+	}{
+		Deeplink: bi.deeplink,
+	}
+
+	mainTmpl, err := template.New("").Parse(`@import UIKit;
 @import Gio;
 
 @interface GioAppDelegate : UIResponder <UIApplicationDelegate>
 @property (strong, nonatomic) UIWindow *window;
+@property (strong, nonatomic) GioViewController *controller;
 @end
 
 @implementation GioAppDelegate
 - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
 	self.window = [[UIWindow alloc] initWithFrame:[[UIScreen mainScreen] bounds]];
-	GioViewController *controller = [[GioViewController alloc] initWithNibName:nil bundle:nil];
-	self.window.rootViewController = controller;
+	self.controller = [[GioViewController alloc] initWithNibName:nil bundle:nil];
+	self.window.rootViewController = self.controller;
 	[self.window makeKeyAndVisible];
 	return YES;
 }
+{{if .Deeplink}}
+- (BOOL)application:(UIApplication *)app openURL:(NSURL *)url options:(NSDictionary<UIApplicationOpenURLOptionsKey, id> *)options {
+	return [self.controller onDeeplink:url.absoluteString];
+}
+{{end}}
```

Comment:
This disabling of code smacks a bit too much like C preprocessor macros. Why not always override the `openURL` method?

Reply:
That was added to make backward compatible, make possible to compile old Gio programs, with new gogio. 

Otherwise, even if deeplink is not set: it will need this function from Gio.

Reply:
Is backwards compatibility a big problem? I worry about code rot if not enough users run this code path.

Otherwise, would it be possible to detect the Gio version and only enable URI opening for newer versions?

Reply:
I think Gio already have some issues with backward-compatibility (https://github.com/gioui/gio-cmd/blob/4128f253e857270fdc59a4e9a9d4b10f454c2935/gogio/androidbuild.go#L252). 

I'm not sure how to implement some "version-detection", also it still require some `{{if .Deeplink}}`, right?

Reply:
I meant using [go/packages](https://pkg.go.dev/golang.org/x/tools/go/packages#Module) for determining the module version of `gioui.org`.

Reply:
I remove the `{{if .Scheme}}`, for now.